src/index.ts

```diff
@@ -76,24 +83,61 @@ app.use(
     )
 )
 
+// calc sha hash based on ip and agent
+const userHash = (req: Request) => {
+    let ip = req.headers['x-forwarded-for'] || req.socket.remoteAddress
+    ip = Array.isArray(ip) ? ip[0] : ip
+    ip = ip || ''
+
+    let agent = req.headers['user-agent']
+    agent = agent || ''
+
+    return crypto
+        .createHash('sha256')
+        .update(ip + agent)
+        .digest('hex')
+}
+
 app.get(
     '/feedback/:episodeId/:feedbackType',
     async (req: Request, res: Response, next: NextFunction) => {
-        //get users ip address
-        const ip = req.headers['x-forwarded-for'] || req.socket.remoteAddress
-        // get users agent
-        const agent = req.headers['user-agent']
-
         const feedbackType = req.params.feedbackType
+        const hash = userHash(req)
+        const episodeId = req.params.episodeId
+        try {
+            await feedbackApi.handleApiGet(episodeId, hash, feedbackType)
+            res.render('feedback.hbs', { episodeId })
+        } catch (err) {
+            next(err)
+        }
+    }
+)
+
+app.post(
+    '/comments/:episodeId',
+    body('email').isEmail(),
+    body('comment')
+        .not()
+        .isEmpty()
+        .trim()
+        .isLength({ min: 3, max: 1000 })
+        .escape(),
```

Comment:
`escape` escapes html tags. I recommend to store the raw text in the database and escape it when it is sent to a frontend. otherwise we cannot use the raw text for other purposes (e.g. send the text in an email)